digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/util/TrafficSignLinearGenerator.scala

```diff
@@ -0,0 +1,682 @@
+        Some("ROADNAME_FI", signRoadLink.attributes("ROADNAME_FI").toString)
+      } else if (signRoadLink.attributes.get("ROADNAME_SE").exists(_.toString.trim.nonEmpty)) {
+        Some("ROADNAME_FI", signRoadLink.attributes("ROADNAME_SE").toString)
+      } else
+        None
+
+    //RoadLink with the same Finnish/Swedish name
+    tsRoadNameInfo.map { case (roadNamePublicIds, roadNameSource) =>
+      roadLinkService.getRoadLinksAndComplementaryByRoadNameFromVVH(roadNamePublicIds, Set(roadNameSource), false).filter(_.administrativeClass != State)
+    }.getOrElse(Seq())
+  }
+
+  def isToUpdateRelation(newSeg: TrafficSignToLinear)(oldSeg: TrafficSignToLinear): Boolean = {
+    oldSeg.roadLink.linkId == newSeg.roadLink.linkId && oldSeg.sideCode == newSeg.sideCode &&
+      Math.abs(oldSeg.startMeasure - newSeg.startMeasure) < 0.01 && Math.abs(oldSeg.endMeasure - newSeg.endMeasure) < 0.01 &&
+      compareValue(oldSeg.value, newSeg.value)
+  }
+
+  def isToUpdateValue(newSeg: TrafficSignToLinear)(oldSeg: TrafficSignToLinear): Boolean = {
+    oldSeg.roadLink.linkId == newSeg.roadLink.linkId && oldSeg.sideCode == newSeg.sideCode &&
+      Math.abs(oldSeg.startMeasure - newSeg.startMeasure) < 0.01 && Math.abs(oldSeg.endMeasure - newSeg.endMeasure) < 0.01
+  }
+
+  def applyChangesBySegments(allSegments: Set[TrafficSignToLinear], existingSegments: Seq[TrafficSignToLinear]) : Unit = {
+    if (allSegments.isEmpty)
+      deleteLinearAssets(existingSegments)
+    else {
+      val segment = allSegments.head
+
+      val toUpdateRelation = existingSegments.filter(isToUpdateRelation(segment))
+      if (toUpdateRelation.nonEmpty) {
+        val head = toUpdateRelation.head
+
+        updateRelation(segment, head)
+        applyChangesBySegments(allSegments.filterNot(_ == segment), existingSegments.filterNot(_ == head))
+
+      } else {
+        val toUpdateValue = existingSegments.filter(isToUpdateValue(segment))
+        if (toUpdateValue.nonEmpty) {
+          val head = toUpdateValue.head
+
+          updateLinearAsset(head.oldAssetId.get, segment.value, userUpdate)
+          updateRelation(segment, head)
+
+          applyChangesBySegments(allSegments.filterNot(_ == segment), existingSegments.filterNot(_ == head))
+        } else {
+          createLinearAssetAccordingSegmentsInfo(segment, userCreate)
+
+          applyChangesBySegments(allSegments.filterNot(_ == segment), existingSegments)
+        }
+      }
+    }
+  }
+
+  def iterativeProcess(roadLinks: Seq[RoadLink], processedRoadLinks: Seq[RoadLink]): Unit = {
+    val roadLinkToBeProcessed = roadLinks.diff(processedRoadLinks)
+
+    if (roadLinkToBeProcessed.nonEmpty) {
+      val roadLink = roadLinkToBeProcessed.head
+      println(s"Processing roadLink linkId ${roadLink.linkId}")
+      val allRoadLinksWithSameName = withDynTransaction {
+        val allRoadLinksWithSameName = getAllRoadLinksWithSameName(roadLink)
```

Comment:
Just to simplify, rename the variable allRoadLinksWithSameName because it's equal to the result variable of this method.